File: src/my_project/easy_problems/from151To200/intersection_of_two_arrays.py
from typing import List 

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:

        set1 = set(nums1)
        set2 = set(nums2)
        answer =  list()
        for num in set1:
            if num in set2:
                answer.append(num)



        return answer 


solution = Solution()

print(solution.intersection(nums1 = [1,2,2,1], nums2 = [2,2]))

print(solution.intersection(nums1 = [4,9,5], nums2 = [9,4,9,8,4]))

File: src/my_project_typescript/easy_problems_typescript/from51To100/intersectionOfTwoArrays.ts
function intersection(nums1: number[], nums2: number[]): number[] {

    let dictionaryOne: any = {}
    let dictionaryTwo: any = {}
    let answer: Array<number> = []

    for (let num of nums1){
        dictionaryOne[num] = 0
    }

    for (let num of nums2){
        dictionaryTwo[num] = 0
    }

    nums1 = []
    nums2 = []
    for (let num in dictionaryOne){
        nums1.push(parseInt(num))
        
    }   
    for (let num in dictionaryTwo){
        nums2.push(parseInt(num))
        
    }

    for (let num of nums1){
        if (nums2.includes(num)){
            answer.push(num)
        }
    }
    
    return answer
};


console.log(intersection([1,2,2,1], [2,2]))

console.log(intersection([4,9,5], [9,4,9,8,4]))
